src/matrix/room/timeline/Timeline.js

```diff
@@ -236,8 +247,50 @@ export class Timeline {
     /** @package */
     addEntries(newEntries) {
         this._addLocalRelationsToNewRemoteEntries(newEntries);
+        this._updateFetchedEntries(newEntries);
```

Comment:
addEntries is not just called for sync, but also for gap fills. So it is possible that an event we previously didn't know about has now been added to the storage and timeline by a gap fill. Should we transition the fetched entry to a remote event entry from the timeline in all of the dependants?

Reply:
Yes we should. Otherwise the dependents of the entry created before gap fill will become dead/forgotten.